identity-wallet/src/state/qr_code/reducers/read_authorization_request.rs

```diff
@@ -92,15 +93,25 @@ pub async fn read_authorization_request(state: AppState, action: Action) -> Resu
             let domain_validation: Box<crate::state::did::validate_domain_linkage::ValidationResult> =
                 Box::new(validate_domain_linkage(url, did).await);
 
+            // This collects the Trusted Issuer domains and checks wether they are valid Url's.
+            // For UX it's not mandatory to add the domain including the http(s):// prefix and the trailing /,
+            // therefore we double check the domain after adding the suffix and prefix.
             let trusted_domains: Vec<String> = state
                 .trust_lists
                 .0
                 .iter()
                 .flat_map(|trust_list| {
-                    trust_list
-                        .entries
-                        .iter()
-                        .filter_map(|(domain, trusted)| trusted.then_some(domain.clone()))
+                    trust_list.entries.iter().filter_map(|(domain, trusted)| {
+                        if *trusted {
+                            let url = Url::parse(domain).unwrap_or_else(|_| {
+                                let formatted_domain = format!("https://{}/", domain.trim_end_matches('/'));
+                                Url::parse(&formatted_domain).expect("Failed to format Trusted Issuer domain as URL")
+                            });
+                            Some(url.to_string())
```

Comment:
I believe your solution is safe, but just to make sure we don't miss any edge-cases that could panic here on the `except` I would suggest to replace this part with something like this:
```rust
                            Url::parse(domain)
                                .ok()
                                .or_else(|| {
                                    let formatted_domain = format!("https://{}/", domain.trim_end_matches('/'));
                                    Url::parse(&formatted_domain).ok()
                                })
                                .map(|url| url.to_string())
```

This will ensure that `None` is returned instead of a panic if the Url-parsing did not work.

Reply:
should be good now